lib/pavilion/test_run.py

```diff
@@ -234,20 +243,10 @@ def __init__(self, pav_cfg, config,
 
         build_config = self.config.get('build', {})
 
-        # make sure build source_download_name is not set without
-        # source_location
-        try:
-            if build_config['source_download_name'] is not None:
-                if build_config['source_location'] is None:
-                    msg = "Test could not be built. Need 'source_location'."
-                    self.status.set(STATES.BUILD_ERROR,
-                                    "'source_download_name is set without a "
-                                    "'source_location'")
-                    raise TestConfigError(msg)
-        except KeyError:
-            # this is mostly for unit tests that create test configs without a
-            # build section at all
-            pass
+        if (build_config.get('source_path') is None and
+                build_config.get('source_url') is not None):
+            raise TestConfigError(
```

Comment:
This is an interesting case.  I guess we're requiring that the test creator know the form that the test will have after being downloaded.  In the case of 'source_download: latest', could we not just determine the 'source_path' based on what we get from downloading 'source_url'?